src/controllers/api/missionInventoryUpdateController.ts

```diff
@@ -50,13 +63,133 @@ const missionInventoryUpdateController: RequestHandler = async (req, res) => {
     try {
         const parsedData = JSON.parse(data) as MissionInventoryUpdate;
         if (typeof parsedData !== "object" || parsedData === null) throw new Error("Invalid data format");
-        await missionInventoryUpdate(parsedData, id);
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+
+        const missionCredits = parsedData.RegularCredits || 0;
+        const creditsBonus = 0;
+        const totalCredits = missionCredits + creditsBonus;
+
+        const MissionCredits = [missionCredits, missionCredits]; // collected credits
+        const CreditsBonus = [creditsBonus, creditsBonus]; // mission reward
+        const TotalCredits = [totalCredits, totalCredits];
+
+        console.log(getRewards(parsedData.RewardInfo));
+
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        const InventoryJson = JSON.stringify(await missionInventoryUpdate(parsedData, id));
+        res.json({
+            // InventoryJson, // this part will reset game data and missions will be locked
+            TotalCredits,
+            CreditsBonus,
+            MissionCredits
+        });
     } catch (err) {
         console.error("Error parsing JSON data:", err);
     }
+};
+
+/*
+**** OUTPUT ****
+- [x]  InventoryJson
+- [ ]  MissionRewards
+- [x]  TotalCredits
+- [x]  CreditsBonus
+- [x]  MissionCredits
+- [ ]  InventoryChanges
+- [ ]  FusionPoints int
+*/
+
+interface StringDictionary {
+    [key: string]: string;
+}
+const getRewards = (rewardInfo: MissionInventoryUpdateRewardInfo | undefined) => {
+    if (!rewardInfo) return;
+
+    const missionName = (missionNames as StringDictionary)[rewardInfo.node];
+    const rewards = missionsDropTable.find(i => i.mission === missionName)?.rewards;
+
+    if (!rewards) return [];
+
+    // TODO - add Rotation logic
+
+    // Separate guaranteed and chance drops
+    const guaranteedDrops: Reward[] = [];
+    const chanceDrops: Reward[] = [];
+    for (const reward of rewards) {
+        if (reward.chance === 100) guaranteedDrops.push(reward);
+        else chanceDrops.push(reward);
+    }
+
+    const randomDrop = getRandomRewardByChance(chanceDrops);
+    if (randomDrop) guaranteedDrops.push(randomDrop);
+
+    return formatRewardsToInventoryType(guaranteedDrops);
+};
+
+interface Reward {
```

Comment:
can you seperate interfaces into seperate files? we got the /types folder, for example string dictionary could be in /types/commonTypes.ts